File: java/gov/nasa/gsfc/gmsec/api/field/StringField.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file StringField.java
 */

package gov.nasa.gsfc.gmsec.api.field;

import gov.nasa.gsfc.gmsec.api.jni.field.JNIStringField;


/**
 * Specialized Field class that can be used to retain a string value.
 */
public class StringField extends Field
{
	/**
	 * This constructor is for internal GMSEC API use only.
	 * @param field A JNIStringField object.
	 */
	public StringField(JNIStringField field)
	{
		setInternal(field);
	}


	/**
	 * Default constructor.
	 *
	 * @param name Name of the field.
	 * @param data Data to associate with the field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name is null, or contains an empty string.
	 * @throws IllegalArgumentException Thrown if the data string is null.
	 */
	public StringField(String name, String data) throws IllegalArgumentException
	{
		validateName(name);

		if (data == null)
		{
			throw new IllegalArgumentException("StringField Data is null");
		}

		setInternal(new JNIStringField(name, data));
	}


	/**
	 * Copy constructor.
	 *
	 * @param other The other StringField object to copy.
	 *
	 * @throws IllegalArgumentException Thrown if the given StringField object is null.
	 */
	public StringField(StringField other) throws IllegalArgumentException
	{
		if (other == null)
		{
			throw new IllegalArgumentException("StringField object is null");
		}

		setInternal(new JNIStringField((JNIStringField) Field.getInternal(other)));
	}


	/**
	 * Returns the data stored within the StringField.
	 *
	 * @return A string value.
	 */
	public String getValue()
	{
		JNIStringField jStringField = (JNIStringField) getInternal();

		return jStringField.getValue();
	}
}

File: framework/src/gmsec5/c/field/string_field.cpp
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/** @file string_field.cpp
 *
 *  @brief This file contains functions for the management of a StringField object.
 */


#include <gmsec5/c/field/string_field.h>

#include <gmsec5/field/StringField.h>

#include <gmsec5/GmsecException.h>
#include <gmsec5/Status.h>


using namespace gmsec::api5;


GMSEC_Field CALL_TYPE stringFieldCreate(const char* fieldName, const char* value, GMSEC_BOOL isHeader, GMSEC_Status status)
{
	StringField* field = NULL;
	Status       result;

	try
	{
		field = new StringField(fieldName, value, isHeader == GMSEC_BOOL::GMSEC_TRUE);
	}
	catch (const GmsecException& e)
	{
		result = Status(e);
	}

	if (status)
	{
		*(reinterpret_cast<Status*>(status)) = result;
	}

	return field;
}


const char* CALL_TYPE stringFieldGetValue(GMSEC_Field field, GMSEC_Status status)
{
	Field*       f           = reinterpret_cast<Field*>(field);
	StringField* stringField = dynamic_cast<StringField*>(f);
	const char*  value       = 0;
	Status       result;

	if (stringField)
	{
		value = stringField->getValue();
	}
	else
	{
		result = Status(FIELD_ERROR, INVALID_FIELD, "Field object is not a StringField");
	}

	if (status)
	{
		*(reinterpret_cast<Status*>(status)) = result;
	}

	return value;
}
